reference-data-proxy/src/v1/controllers/acbs.controller.ts

```diff
@@ -103,30 +131,63 @@ export const issueAcbsFacilityPOST = async (req: Request, res: Response) => {
   return res.status(400).send();
 };
 
-const createAcbsRecord = async (deal: any, bank: any) => {
-  if (deal) {
+/**
+ * Invoked Azure DOF using HTTP `POST` method.
+ * @param {Object} amendment Amendment object comprising facility ID and amends. A amendment at a time is processed.
+ * @returns {Object} DOF Response
+ */
+const amendAcbsFacility = async (amendment: object) => {
+  const hasAcceptablePayload = Object.prototype.hasOwnProperty.call(amendment, 'coverEndDate') || Object.prototype.hasOwnProperty.call(amendment, 'amount');
+
+  if (amendment && hasAcceptablePayload) {
     const response = await axios({
       method: 'post',
-      url: `${acbsFunctionUrl}/api/orchestrators/acbs`,
+      url: `${acbsFunctionUrl}/api/orchestrators/acbs-amend-facility`,
       data: {
-        deal,
-        bank,
+        amendment,
       },
-    }).catch((err: any) => err);
+    }).catch((e: any) => e);
+
     return response;
   }
   return {};
 };
 
-export const createAcbsRecordPOST = async (req: Request, res: Response) => {
+/**
+ * ACBS facility amendment entry function.
+ * Constructs acceptable payload by DOF.
+ * @param {Object} req Request
+ * @param {Object} res Response
+ * @return {Object} Response object with HTTP code as `status` and response as `data`.
+ */
+export const amendAcbsFacilityPost = async (req: Request, res: Response) => {
   if (req) {
-    try {
-      const { deal, bank } = req.body;
-      const { status, data } = await createAcbsRecord(deal, bank);
-      return res.status(status).send(data);
-    } catch (error) {
-      console.error('ACBS post failed ', { error });
-    }
+    const { id } = req.params;
+    const { amendments, deal } = req.body;
+    // Construct payload
+    const payload = {
+      facilityId: id,
+      amount: amendments.ukefExposure,
+      coverEndDate: amendments.coverEndDate,
+      deal,
+    };
+
+    // Refine payload
+
+    // Change requested
+    const { changeFacilityValue, changeCoverEndDate } = amendments;
+    // UW Decision
+    const { value, coverEndDate } = amendments.ukefDecision;
+    const valueDeclined = value === UNDERWRITER_MANAGER_DECISIONS.DECLINED;
+    const coverEndDateDeclined = coverEndDate === UNDERWRITER_MANAGER_DECISIONS.DECLINED;
+    // Delete frivolous property
+    if (!changeFacilityValue || valueDeclined) delete payload.amount;
```

Comment:
can we use something like this instead?
```
if (!changeFacilityValue || value === UNDERWRITER_MANAGER_DECISIONS.DECLINED) {
    delete payload.amount;
}

```
we don't really write the `if` statements in online - we tend to be explicit even if it expands on multiple lines

Reply:
Have re-introduced `if` branch.
Have left the variable in place to keep it consistent and future proof.